core/citrus-base/src/test/java/org/citrusframework/report/TestResultsTest.java

```diff
@@ -18,58 +18,122 @@
 
 import org.citrusframework.TestResult;
 import org.citrusframework.exceptions.CitrusRuntimeException;
-import org.testng.Assert;
+import org.testng.annotations.BeforeMethod;
 import org.testng.annotations.Test;
 
+import java.time.Duration;
+import java.util.concurrent.BrokenBarrierException;
+import java.util.concurrent.CyclicBarrier;
+
+import static java.lang.Thread.currentThread;
+import static java.util.concurrent.Executors.newFixedThreadPool;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
 /**
  * @author Christoph Deppisch
  */
 public class TestResultsTest {
 
+    private TestResults fixture;
+
+    @BeforeMethod
+    public void beforeMethodSetup() {
+        fixture = new TestResults();
+    }
+
+    @Test
+    public void addResultIsThreadSafe() throws InterruptedException {
+        var numberOfThreads = 10;
+        var executorService = newFixedThreadPool(numberOfThreads);
+        var barrier = new CyclicBarrier(numberOfThreads);
+        var durationToAdd = Duration.ofMillis(100);
+
+        for (int i = 0; i < numberOfThreads; i++) {
+            final int index = i;
+            executorService.submit(() -> {
+                try {
+                    // Ensure all threads start at the same time
+                    barrier.await();
+
+                    var testResult = mock(TestResult.class);
+
+                    doAnswer((invocation) -> {
+                        // Simulating a long duration modification by adding a sleep
+                        Thread.sleep(index * 10);
+                        return durationToAdd;
+                    }).when(testResult).getDuration();
+
+                    fixture.addResult(testResult);
+                } catch (InterruptedException | BrokenBarrierException e) {
+                    currentThread().interrupt();
+                }
+            });
+        }
+
+        executorService.shutdown();
```

Comment:
This should be done in a finally block, to ensure in any case, that the service is shutdown:

Something like:
```java
} finally {
        if (!executorService.isTerminated()) {
            List<Runnable> unfinishedTasks = executorService.shutdownNow();
            if (!unfinishedTasks.isEmpty()) {
                fail("Executor did not terminate all tasks upon shutdown.");
            }
        }
    }
```